bug-finding.ipynb

```diff
@@ -196,42 +246,35 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 8,
+   "execution_count": 7,
    "metadata": {},
    "outputs": [
     {
      "data": {
       "text/plain": [
-       "0      NaN\n",
-       "1      NaN\n",
-       "2      NaN\n",
-       "3      NaN\n",
-       "4      NaN\n",
-       "        ..\n",
-       "3165   NaN\n",
-       "3166   NaN\n",
-       "3167   NaN\n",
-       "3168   NaN\n",
-       "3169   NaN\n",
-       "Name: tokens, Length: 3170, dtype: float64"
+       "26932"
       ]
      },
-     "execution_count": 8,
+     "execution_count": 7,
      "metadata": {},
      "output_type": "execute_result"
     }
    ],
    "source": [
-    "observed_freq_the = df['tokens'].str.find('the')\n",
-    "\n",
+    "observed_freq_the = len([t for t in df[\"nlp_docs\"].explode() if t.lemma_ == 'the'])\n",
     "observed_freq_the"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# Bug 2 analysis and response"
+    "# Bug 2 analysis and response\n",
+    "\n",
+    "In the above code, using the str.find() method is not applicable. The str.find() method in Python is used to locate the first occurrence of a specified substring within a string. Additionally, the way the code is written is not designed to give you the number of occurences of the token \"the\" and just returns a series instead. The way to correct this is to use list comprehension and compute the length of the list where the lemma is equal to \"the\".\n",
+    "\n",
+    "One way to do this is:\n",
+    "len([t for t in df[\"nlp_docs\"].explode() if t.lemma_ == 'the'])\n"
```

Comment:
Good! A bonus when using the lemma instead of just counting tokens is that you're accounting for potential capitalization differences (e.g., when "The" begins a sentence).